src/software/ai/hl/stp/tactic/stop/stop_tactic_test.cpp

```diff
@@ -18,3 +30,57 @@ TEST(StopTacticTest, test_calculate_robot_cost)
     // equally
     EXPECT_EQ(0.5, tactic.calculateRobotCost(robot, world));
 }
+
+TEST_F(StopTacticTest, robot_already_stopped)
+{
+    BallState ball_state(Point(0, 0.5), Vector(0, 0));
+
+    auto friendly_robots = TestUtil::createMovingRobotStatesWithId(
+        {Point(-3, 2.5), Point()}, {Vector(), Vector()});
+    auto enemy_robots = TestUtil::createStationaryRobotStatesWithId({Point(4, 0)});
+
+    auto tactic = std::make_shared<StopTactic>(false);
+    setTactic(tactic);
+    setFriendlyRobotId(1);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            for (unsigned i = 0; i < 1000; i++)
+            {
+                robotHalt(world_ptr, yield);
+            }
```

Comment:
why  not just make this a non-terminating validation function?

Reply:
because the robots might start the test moving but we want to check that they eventually stop